packages/cli/src/workflows/workflow.service.ts

```diff
@@ -220,31 +220,39 @@ export class WorkflowService {
 		return updatedWorkflow;
 	}
 
-	async delete(user: User, workflowId: string): Promise<WorkflowEntity | undefined> {
+	async delete(
+		user: User,
+		workflowId: string,
+		em?: EntityManager,
+	): Promise<WorkflowEntity | undefined> {
+		em = em ?? this.workflowRepository.manager;
 		await this.externalHooks.run('workflow.delete', [workflowId]);
 
-		const workflow = await this.sharedWorkflowRepository.findWorkflowForUser(workflowId, user, [
-			'workflow:delete',
-		]);
+		const workflow = await this.sharedWorkflowRepository.findWorkflowForUser(
+			workflowId,
+			user,
+			['workflow:delete'],
+			{ em },
+		);
 
 		if (!workflow) {
 			return;
 		}
 
 		if (workflow.active) {
 			// deactivate before deleting
-			await this.activeWorkflowRunner.remove(workflowId);
+			await this.activeWorkflowRunner.remove(workflowId, em);
```

Comment:
I'm thinking whether from this point downstream we don't use the transactions and leave as regular queries, as we're getting the `entityManager` to places it's not supposed to be WDYT?

I'm ok with passing EM across repositories, but this seems to be going too far and would make it impossible to trace back where a given EM came from.

Reply:
WDYT @valya ?

Reply:
Decision: 

`WorkflowService.delete`'s body is split out into 2 new functions:
1. deactivate workflow
	* does not take the `em`
2. delete workflow
	* does take an optional `em`

Then `ProjectService.delete` can call 1. without passing the `em` and then call 2. with passing the `em`.

That way we don't have to pass the `em` around as much. 

This leaves a race condition in the code that already exists right now: The workflow could be activated again between it being deactivated and it being deleted. 
This race condition exists already and we're ok with keeping in, given that the time between the deactivation and deletion is neglectable and it will probably never happen.